### PR TITLE
getBidPriceAsync include address information

### PR DESCRIPTION
**Pull request by richardliang**

### CODE REVIEW

File: src/types/common.ts [hunk 1]
```diff
@@ -83,6 +83,11 @@ export interface TokenFlows {
   outflow: BigNumber[];
 }
 
+export interface TokenFlowsDetails {
```
Comment (asoong) [inline src/types/common.ts]: Nice

File: src/api/RebalancingAPI.ts [hunk 2]
```diff
@@ -184,10 +184,41 @@ export class RebalancingAPI {
...
+    rebalancingSetTokenAddress: Address,
+    bidQuantity: BigNumber,
+  ): Promise<TokenFlowsDetails> {
     await this.assertGetBidPrice(rebalancingSetTokenAddress, bidQuantity);
 
-    return await this.rebalancingSetToken.getBidPrice(rebalancingSetTokenAddress, bidQuantity);
+    const tokenFlowComponents = await this.rebalancingSetToken.getBidPrice(rebalancingSetTokenAddress, bidQuantity);
+    const tokenAddresses = await this.rebalancingSetToken.getCombinedTokenArray(rebalancingSetTokenAddress);
+
+    const inflow = tokenFlowComponents.inflow.reduce((acc, unit, index) => {
```
Comment (asoong) [inline src/api/RebalancingAPI.ts]: what's `acc`? spell out here

File: src/api/RebalancingAPI.ts [hunk 3]
```diff
@@ -184,10 +184,41 @@ export class RebalancingAPI {
...
+      const bigNumberUnit = new BigNumber(unit);
```
Comment (asoong) [inline src/api/RebalancingAPI.ts]: the return value from the contract wrapper is already a big number I believe

File: test/integration/api/RebalancingAPI.spec.ts [hunk 4]
```diff
@@ -1058,12 +1058,36 @@ describe('RebalancingAPI', () => {
           DEFAULT_CONSTANT_AUCTION_PRICE
         );
 
+        const expectedTokenAddresses = await rebalancingSetToken.getCombinedTokenArray.callAsync();
+
+        const expectedInflowAddressArray = expectedTokenFlowArrays['inflow'].reduce((acc, unit, index) => {
```
Comment (asoong) [inline test/integration/api/RebalancingAPI.spec.ts]: Depending on your test case, I think it's more effective to spell out what the expected values are for your particular set up parameters instead of recreating the same logic here which would always produce a passing test. e.g. `expectedOutflowAddressArray = [ { ... } ]`

File: test/integration/api/RebalancingAPI.spec.ts [hunk 5]
```diff
@@ -1058,12 +1058,36 @@ describe('RebalancingAPI', () => {
...
+        }, []);
+
         const returnedInflowArray = JSON.stringify(returnedTokenFlowArrays['inflow']);
-        const expectedInflowArray = JSON.stringify(expectedTokenFlowArrays['inflow']);
+        const expectedInflowArray = JSON.stringify(expectedInflowAddressArray);
         expect(returnedInflowArray).to.eql(expectedInflowArray);
 
         const returnedOutflowArray = JSON.stringify(returnedTokenFlowArrays['outflow']);
-        const expectedOutflowArray = JSON.stringify(expectedTokenFlowArrays['outflow']);
+        const expectedOutflowArray = JSON.stringify(expectedOutflowAddressArray);
```
Comment (asoong) [inline test/integration/api/RebalancingAPI.spec.ts]: Add a separate test case where one of the components has zero units and verify that it is excluded

**Comment by asoong** [comment]
Looks good so far, don't forget to bump the version #

File: src/api/RebalancingAPI.ts (hunk 3 above)
Comment (richardliang) [reply src/api/RebalancingAPI.ts]: apparently the contract wrapper converts it into string?

TypeError: unit.gt is not a function

      196 |     const inflow = tokenFlowComponents.inflow.reduce((accumulator, unit, index) => {
      197 |       const bigNumberUnit = new BigNumber(unit);
    > 198 |       if (unit.gt(0)) {
      199 |         accumulator.push({
      200 |           address: tokenAddresses[index],
      201 |           unit,

File: test/integration/api/RebalancingAPI.spec.ts [hunk 6]
```diff
@@ -1049,24 +1050,84 @@ describe('RebalancingAPI', () => {
...
+          ) => {
+          const bigNumberUnit = new BigNumber(component.unit);
+          if (bigNumberUnit.eq(0)) {
+            accumulator++;
+          }
+          return accumulator;
+        }, 0);
+
+        // Ensure there are inflow / outflow components greater than 0
+        expect(returnedInflowZeroCount).to.greaterThan(0);
```
Comment (asoong) [inline test/integration/api/RebalancingAPI.spec.ts]: Can you be precise about how many there are for these particular bid parameters instead of greater than?

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "setprotocol.js",
-  "version": "1.1.1-rc3",
+  "version": "1.1.1-rc4",
```
Comment (asoong) [inline package.json]: Don't forget to update the README.md
